src/Config/ConfigUpdater.php

```diff
@@ -0,0 +1,276 @@
+      }
+
+      $succeeded[$config_name] = $this->t('@config was successfully imported', ['@config' => $config_name]);
+    }
+
+    return new ConfigUpdateResult($succeeded, $failed);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function revert(array $config_names, $skip_modified = TRUE) {
+    $succeeded = [];
+    $failed = [];
+
+    foreach ($config_names as $config_name) {
+      // Read the config from the file.
+      $value = $this->loadFromExtension($config_name);
+      if (!$value) {
+        $failed[$config_name] = $this->t('@config did not exist in extension storage', ['@config' => $config_name]);
+        continue;
+      }
+
+      // Check the configuration object's hash and see if it has been modified.
+      if ($this->isModified($config_name) && $skip_modified) {
+        $failed[$config_name] = $this->t('@config has been modified and was not reverted', ['@config' => $config_name]);
+        continue;
+      }
+
+      $type = $this->getConfigType($config_name);
+      if ($type == 'system.simple') {
+        // Load the current config and replace the value.
+        $this->configFactory->getEditable($config_name)->setData($value)->save();
+      }
+      else {
+        // Load the current config entity and replace the value, with the
+        // old UUID.
+        $definition = $this->entityTypeManager->getDefinition($type);
+        $id_key = $definition->getKey('id');
+
+        $id = $value[$id_key];
+        /** @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $entity_storage */
+        $entity_storage = $this->entityTypeManager->getStorage($type);
+        /** @var \Drupal\Core\Config\Entity\ConfigEntityInterface $entity */
+        $entity = $entity_storage->load($id);
+        $uuid = $entity->get('uuid');
+        $entity = $entity_storage->updateFromStorageRecord($entity, $value);
+        $entity->set('uuid', $uuid);
+        $entity->save();
+      }
+
+      $succeeded[$config_name] = $this->t('@config was successfully reverted', ['@config' => $config_name]);
+    }
+
+    return new ConfigUpdateResult($succeeded, $failed);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function delete(array $config_names) {
+    $succeeded = [];
+    $failed = [];
+
+    foreach ($config_names as $config_name) {
+      $value = $this->loadFromActive($config_name);
+      if (!$value) {
+        $failed[$config_name] = $this->t('@config did not exist in extension storage', ['@config' => $config_name]);
+        continue;
+      }
+
+      // Check the configuration object's hash and see if it has been modified.
+      if ($this->isModified($config_name)) {
+        $failed[$config_name] = $this->t('@config has been modified and was not deleted', ['@config' => $config_name]);
+        continue;
+      }
+
+      $type = $this->getConfigType($config_name);
+      if ($type == 'system.simple') {
+        $config = $this->configFactory->getEditable($config_name);
+        if (!$config) {
```

Comment:
getEditable() always returns a config object, so the if (!$config) will never work. Furthermore, we already checked active storage above.